openapi-v1.yaml

```diff
@@ -2632,6 +2632,113 @@ definitions:
+      path:
+        description: A slash separated string of group names from the root to this group. Analogous to an OS filepath
+        type: string
+      description:
+        description: A human readable description of the content of the group
+        type: string
+        x-omitempty: true
+    example:
+      id: 0000-0001-0002-0003
+      namespace: my-organization
+      name: intro-to-genomics
+      path: /genomics-course/intro-to-genomics
+      description: contains arrays and notebooks for an 101 course on genomics with TileDB
+
+  GroupMember:
+    description: An asset of a group i.e an array, a udf etc but not a subgroup. These are handled and listed separately.
+    type: object
+    properties:
+      namespace:
+        description: The namespace of the asset. Groups can contain assets from multiple namespaces
+        type: string
+      name:
+        description: The name of the asset
+        type: string
+    example:
+      namespace: tiledb
+      name: median
+
+  GroupCreate:
+    description: Initial attributes for the creation of a new group
+    type: object
+    properties:
+      parent:
+        description: The name of the parent of the group. If empty, then the new group is a top level group.
+        type: string
+      name:
+        description: The name of the group. It is unique within the namespace. No 2 groups can have the same name
+        type: string
+      description:
+        description: A human readable description of the content of the group
+        type: string
+    example:
+      parent: genomics-course
+      name: intro-to-genomics
+      description: contains arrays and notebooks for an 101 course on genomics with TileDB
+
+  GroupUpdate:
+    description: Updates for a group. New values for the attributes.
+    type: object
+    properties:
+      name:
+        description: The name of the group. It is unique within the namespace. No 2 groups can have the same name
+        type: string
+      description:
+        description: A human readable description of the content of the group
+        type: string
+    example:
+      name: intro-to-genomics
+      description: contains arrays and notebooks for an 101 course on genomics with TileDB
+
+  GroupListing:
+    description: The contents of a groups i.e attributes, subgroups and assets
+    allOf:
+      - $ref: '#/definitions/GroupAttributes'
+      - type: object
+        properties:
+          groups:
+            description: The attributes of the subgroups
+            type: array
+            x-omitempty: true
+            items:
+              $ref: "#/definitions/GroupAttributes"
+          assets:
+            description: One page of assets of the group as ArrayInfos
+            type: array
+            x-omitempty: true
+            items:
+              $ref: "#/definitions/ArrayInfo"
+          pagination_metadata:
+            $ref: "#/definitions/PaginationMetadata"
+
+  GroupListingTopLevel:
```

Comment:
Can we concatenate `GroupListingTopLevel` and `GroupListing`?

Reply:
Yes. The user calls another method `listTopLevelGroups` so he knows that some attributes of `GroupListing` for top level have empty values